cpp/CMakeLists.txt

```diff
@@ -73,7 +73,6 @@ endif()
 
 set(BUILD_SUPPORT_DIR "${CMAKE_SOURCE_DIR}/build-support")
 
-set(CLANG_FORMAT_VERSION "6.0")
```

Comment:
Yes this is used (see cmake_modiles/FindClangTools.cmake), so we can't remove this

Reply:
Its definition prevents us from hitting the the `else` statement in that file (https://github.com/apache/arrow/blob/master/cpp/cmake_modules/FindClangTools.cmake#L97) and as a result only allows the build script to find the literal binary `clang-format-6`.

Reply:
I'll remove it from the PR since it looks like CI fails without it, but that looks like a bug to me.  I will file an issue.  We ought to be passing that as an optional argument to CMake so we're able to use more toolchains in development than just LLVM 6.

Reply:
We are pinned to clang-format-6.0 for development because the formatting changes slightly from version to version. So if you used a different version to format the codebase, our CI would reject the changes.

Reply:
Got it. Thanks.